src/functional/fencil_processors/source_modules/source_modules.py

```diff
@@ -15,11 +15,37 @@
 
 
 from dataclasses import dataclass
-from typing import Sequence
+from typing import Generic, Protocol, TypeVar
 
 import numpy
 
 
+class SupportedLanguage(Protocol):
+    """Ensures consistent code formatting along the pipeline."""
+
+    @property
+    def name(self) -> str:
+        ...
+
+    def format_source(self, source_code: str) -> str:
+        ...
+
+
+class LanguageWithHeaders(SupportedLanguage, Protocol):
+    """Ensures consistent file naming for languages that split code into include (header) and implementation files."""
+
+    @property
+    def implementation_extension(self) -> str:
+        ...
+
+    @property
+    def include_extension(self) -> str:
+        ...
```

Comment:
Same reply as above. If someone needs all the headers to be called `.my.custom.hpp` for their integration purposes, they would need to implement a new (otherwise redundant) class.